crucible/src/Lang/Crucible/Backend.hs

```diff
@@ -378,58 +379,61 @@ ppLoc :: ProgramLoc -> PP.Doc ann
 ppLoc l = PP.pretty (plSourceLoc l)
 
 type IsSymInterface sym =
-  ( IsBoolSolver sym
-  , IsSymExprBuilder sym
+  ( IsSymExprBuilder sym
   , IsInterpretedFloatSymExprBuilder sym
   )
 
 -- | This class provides operations that interact with the symbolic simulator.
 --   It allows for logical assumptions/assertions to be added to the current
 --   path condition, and allows queries to be asked about branch conditions.
-class IsBoolSolver sym where
+class IsBoolSolver sym bak | bak -> sym where
```

Comment:
I would find it helpful to have the Haddocks clarify what exactly the `bak` type variable represents, as well as the relationship between `bak` and `sym`.